sdk/core/Azure.Core/src/Shared/PemReader.cs

```diff
@@ -83,7 +83,17 @@ public static X509Certificate2 LoadCertificate(ReadOnlySpan<char> data, byte[] c
             {
                 if (allowCertificateOnly)
                 {
+#if NET9_0_OR_GREATER
```

Comment:
[SYSLIB0057: X509Certificate2 and X509Certificate constructors for binary and file content are obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0057)